vmm/src/device_tree.rs

```diff
@@ -165,9 +165,12 @@ mod tests {
             aver_eq!(tb, node.id, id2);
 
             // Check remove()
-            let node = device_tree.remove(&id).unwrap();
-            aver_eq!(tb, node.id, id2);
-            aver_eq!(tb, device_tree.0.len(), 0);
+            #[cfg(feature = "pci_support")]
+            {
+                let node = device_tree.remove(&id).unwrap();
+                aver_eq!(tb, node.id, id2);
+                aver_eq!(tb, device_tree.0.len(), 0);
+            }
```

Comment:
As suggested here: https://github.com/cloud-hypervisor/cloud-hypervisor/pull/1324, we should simply remove the `pci_support` from the `remove()` function. This should simplify things for you wrt `device_tree.rs`.

Reply:
Thanks @sboeuf ! This is a really nice solution and it turned out that this solution works pretty well. I have updated the code. Would you mind take a look again? Thanks!